dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java

```
@@ -14,93 +14,116 @@
+            return new StringMessage("hello world");
         });
-        Exchangers.bind(URL.valueOf("dubbo://localhost:" + port), dispatcher);
+        exchangeServer = Exchangers.bind(URL.valueOf("dubbo://localhost:" + port), dispatcher);
     }
 
-    static void dataPackageTest(int port) throws Exception {
+
+    @Test
+    public void testDataPackage() throws Exception {
         ExchangeChannel client = Exchangers.connect(URL.valueOf("dubbo://localhost:" + port));
         Random random = new Random();
         for (int i = 5; i < 100; i++) {
             StringBuilder sb = new StringBuilder();
             for (int j = 0; j < i * 100; j++)
-                sb.append("(" + random.nextLong() + ")");
-            Main.Data d = new Main.Data();
+                sb.append("(").append(random.nextLong()).append(")");
+            Data d = new Data();
             d.setData(sb.toString());
-            client.request(d).get();
+            Assertions.assertEquals(client.request(d).get().toString(), "hello world");
         }
-        System.out.println("send finished.");
+        clients.put(Thread.currentThread().getName(), client);
     }
 
-    static void mutliThreadTest(int tc, final int port) throws Exception {
-        Executor exec = Executors.newFixedThreadPool(tc);
+
+    @Test
+    public void testMutliThread() throws Exception {
+        int tc = 10;
+        ExecutorService exec = Executors.newFixedThreadPool(tc);
         for (int i = 0; i < tc; i++)
-            exec.execute(new Runnable() {
-                public void run() {
-                    try {
-                        test(port);
-                    } catch (Exception e) {
-                        e.printStackTrace();
-                    }
+            exec.execute(() -> {
+                try {
+                    clientExchangeInfo(port);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

for any exception, we should consider this unit test fail.
You can throw this exception or invoke `fail` method in junit library